### DISCUSSION TITLE
Boundary condition ambiguity

### DISCUSSION THREAD

**Discussion by JahanzebTariq** [PMEAL/OpenPNM#2679] [Q&A]
Greetings professor @jgostick and @ma-sadeghi .. I' completed almost all of my work thanks to your help. the only thing I'm concerned and wanted to confirm from you is the boundary conditions.
![image](https://user-images.githubusercontent.com/99289270/215310033-221e7e97-9e7d-4809-95ff-d7eca07c21e3.png)
please tell me are these correct values shown, as the rate bc is 0... please do correct me

**Comment by jgostick** [comment]
The above output is telling you that you have assigned 0 rate boundary conditions and 1757 value conditions. If you want the ``rate`` at a boundary, then you do ``alg.rate(pores=pn.pores('left'), mode='group')``, of course using the algorithm and network names that you used and replacing 'left' with whichever pores you applied inlet (or outlet) bcs to.

**Reply by JahanzebTariq** [reply]
@jgostick .. dear sir I am confused about the meaning of 'rate' and 'value' boundary conditions.. please explain these. I went through the coding examples but i think I didn't get them. I still very confused that why are we calling them rate and value boundary conditions. what sort of boundary conditions are these. what does rate represent..? what does value represent..? please do answer these questions.. thank you

**Reply by jgostick** [reply]
rate is analogous but not quite the same as a "neumann" condition, while value is identical to a "dirichlet" condition.  In other words when you set a rate BC the solver finds the 'value' (e.g. concentration or pressure) needed in the pore to satisfy the requested rate, while if you set the a value bc the solver uses finds the rate leave the pore.

**Reply by JahanzebTariq** [reply]
![image](https://user-images.githubusercontent.com/99289270/215642123-00da4912-1e42-41d4-9822-feeccd600b36.png)
 @jgostick .. in my case I've applied value boundary condition right..? and did not employed rate boundary conditions, because of which the **rate boundary condition value is 0 in the previous image** .. 
![image](https://user-images.githubusercontent.com/99289270/215642778-7274fe00-7a20-4706-9cba-3b0cc14934e5.png)
but as can be scene I have applied rate boundary condition at the inlet when calculation flow rate... please address my issue.. thank you

**Reply by jgostick** [reply]
sf.rate asks the algorithm to find the rate, not set it.

**Reply by jgostick** [reply]
It just occurred to me that this method might be better called ``get_rate``. Obviously much clearer and provides a counter-point to ``set_rate_BC``.  Along those lines, I wonder if we should also have ``get_values``?  This would reaching into the quantify array and return the values.  Probably over thinking it.
